src/driver-steelseries.c

```diff
@@ -440,98 +508,84 @@ steelseries_write_led_v2(struct ratbag_led *led)
 {
 	struct ratbag_device *device = led->profile->device;
 	uint8_t buf[STEELSERIES_REPORT_SIZE] = {0};
-	uint16_t duration, min_duration;
 	int ret;
 
-	buf[0] = STEELSERIES_ID_LED;
-	buf[2] = led->index;
+	struct steelseries_cycle cycle;
+	struct steelseries_point point1;
+	struct steelseries_point point2;
+	struct steelseries_point point3;
+	struct steelseries_point point4;
 
-	/* not sure if these two are needed */
-	buf[15] = 0x01;
-	buf[17] = 0x01;
+	create_cycle(&cycle);
+	cycle.led_id = led->index;
 
 	switch(led->mode) {
 	case RATBAG_LED_OFF:
-		buf[19] = 0x01;
-		buf[27] = 0x01;
-		buf[28] = buf[31] = 0x00;
-		buf[29] = buf[32] = 0x00;
-		buf[30] = buf[33] = 0x00;
-		/* not sure why the duration is set for the steady color or why
-		   it is different for the two LEDs */
-		duration = led->index == 0 ? 10000 : 5000; /* 0x1027 or 0x8813 */
+		point1.color.red = 0x00;
+		point1.color.green = 0x00;
+		point1.color.blue = 0x00;
```

Comment:
A good and easy-to-read solution here would be:
````
const struct ratbag_color black  = { 0x00 };
point1.color = black;
````
and then repeat that for the other cases. Makes it immediately obvious what things are set to.

Reply:
Will do.